server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -847,7 +848,7 @@ public XContentBuilder toInnerXContent(XContentBuilder builder, Params params) t
             }
             // _ignored is the only multi-valued meta field
             // TODO: can we avoid having an exception here?
-            if (field.getName().equals(IgnoredFieldMapper.NAME)) {
+            if (IgnoredFieldMapper.NAME.equals(field.getName()) || IgnoredSourceFieldMapper.NAME.equals(field.getName())) {
```

Comment:
This is breaking some tests anyway.

Reply:
Which tests are failing?

Reply:
SearchResponseTests#testFromXContent
SearchHitTests#testFromXContentLenientParsing
SearchHitTests#testFromXContent
SearchHitsTest#testFromXContent (NOTE this is *HITS* not *HIT*)